### DISCUSSION TITLE
change retention time of topic

### DISCUSSION THREAD

**Discussion by therus000** [strimzi/strimzi-kafka-operator#7701] [Q&A]
good day i upload Kafka to my Kubernetes.
is there an easy way to change the retention time of a topic without redeploy setting?
From ui i didn't find. There is only clear message - and delete topic
one more question about topics. what best way to add user and permission to topics
please dont mad. im new kafka user )))

**Comment by scholzj** [comment]
That depends how you manage your topics. You can just change it using the Kafka Admin APIs or by editing the KafkaTopic custom resource if you use the Topic Operator. In the topic configuration, you can basically use the option https://kafka.apache.org/documentation/#topicconfigs_retention.ms to change it. You can also use https://kafka.apache.org/documentation/#brokerconfigs_log.retention.ms in the broker configuration (in the Kafka custom resource) to change the default for all topics.
